cachix/src/Cachix/Client/Store.hs

```diff
@@ -131,6 +133,27 @@ validPathInfoNarHash vpi =
         *{ strdup((*$fptr-ptr:(refValidPathInfo* vpi))->narHash.to_string().c_str()) }
       |]
 
+-- | Deriver field of a ValidPathInfo struct. Source: store-api.hh
+validPathInfoDeriver :: ForeignPtr (Ref ValidPathInfo) -> IO ByteString
+validPathInfoDeriver vpi =
+  unsafePackMallocCString
+    =<< [C.throwBlock| const char*
+        {
+          std::optional<Path> deriver = (*$fptr-ptr:(refValidPathInfo* vpi))->deriver;
+          return strdup((deriver == "" ? "unknown-deriver" : deriver->c_str()));
+        }
+      |]
+
+-- | References field of a ValidPathInfo struct. Source: store-api.hh
+validPathInfoReferences :: ForeignPtr (Ref ValidPathInfo) -> IO PathSet
+validPathInfoReferences vpi = do
+  ptr <-
+    [C.exp| const PathSet*
+            { &((*$fptr-ptr:(refValidPathInfo* vpi))->references) }
```

Comment:
The `PathSet` wrapper assumes responsibility for `delete`ing its input, so we must copy it to avoid a double free. Something like:

```suggestion
            { new PathSet((*$fptr-ptr:(refValidPathInfo* vpi))->references) }
```